src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/AvailabilityResourceIntegrationTest.kt

```diff
@@ -0,0 +1,210 @@
+        ),
+      ),
+    )
+
+    val availabilityResponse = webTestClient.availabilityCheck(bookingCreator, availabilityRequest)
+
+    assertThat(availabilityResponse).isNotNull
+    with(availabilityResponse) {
+      assertThat(availabilityOk).isFalse()
+      assertThat(alternatives).isNotEmpty()
+      assertThat(alternatives).hasSize(3)
+    }
+  }
+
+  @Test
+  fun `should confirm availability for a probation booking when the prison room is free`() {
+    val bookingCreator = "BOOKING_CREATOR"
+
+    videoBookingRepository.findAll() hasSize 0
+
+    val availabilityRequest = AvailabilityRequest(
+      bookingType = BookingType.PROBATION,
+      courtOrProbationCode = "BLKPPP",
+      prisonCode = WERRINGTON,
+      date = LocalDate.now().plusDays(1),
+      mainAppointment = LocationAndInterval(
+        werringtonLocation.key,
+        Interval(
+          start = LocalTime.of(12, 0),
+          end = LocalTime.of(12, 30),
+        ),
+      ),
+    )
+
+    val availabilityResponse = webTestClient.availabilityCheck(bookingCreator, availabilityRequest)
+
+    assertThat(availabilityResponse).isNotNull
+    with(availabilityResponse) {
+      assertThat(availabilityOk).isTrue()
+      assertThat(alternatives).isEmpty()
+    }
+  }
+
+  @Test
+  fun `should return alternatives for a probation booking when the prison room is already occupied`() {
+    val bookingCreator = "BOOKING_CREATOR"
+
+    videoBookingRepository.findAll() hasSize 0
+
+    prisonSearchApi().stubGetPrisoner("A1111AA", WERRINGTON)
+    locationsInsidePrisonApi().stubPostLocationByKeys(setOf(werringtonLocation.key), WERRINGTON)
+
+    val probationBookingRequest = probationBookingRequest(
+      probationTeamCode = "BLKPPP",
+      probationMeetingType = ProbationMeetingType.PSR,
+      videoLinkUrl = "https://probation.videolink.com",
+      prisonCode = WERRINGTON,
+      prisonerNumber = "A1111AA",
+      startTime = LocalTime.of(9, 0),
+      endTime = LocalTime.of(9, 30),
+      appointmentType = AppointmentType.VLB_PROBATION,
+      location = werringtonLocation,
+    )
+
+    webTestClient.createBooking(bookingCreator, probationBookingRequest)
+
+    videoBookingRepository.findAll() hasSize 1
+
+    // Check for availability of the times just created i.e. an overlapping appointment
+    val availabilityRequest = AvailabilityRequest(
+      bookingType = BookingType.PROBATION,
+      courtOrProbationCode = "BLKPPP",
+      prisonCode = WERRINGTON,
+      date = LocalDate.now().plusDays(1),
+      mainAppointment = LocationAndInterval(
+        werringtonLocation.key,
+        Interval(
+          start = LocalTime.of(9, 0),
+          end = LocalTime.of(9, 30),
+        ),
+      ),
+    )
+
+    val availabilityResponse = webTestClient.availabilityCheck(bookingCreator, availabilityRequest)
+
+    assertThat(availabilityResponse).isNotNull
+    with(availabilityResponse) {
+      assertThat(availabilityOk).isFalse()
+      assertThat(alternatives).isNotEmpty()
```

Comment:
Again don't really need the `isNotEmpty`.

Reply:
Done